src/NameRegistry.sol

```diff
@@ -44,6 +44,26 @@ contract NameRegistry is
         address destination;
     }
 
+    /**
+     * @dev RegistrationMetadata struct represents the fname registration data.
+     * @param recoveryOf The address that can recover the fname.
+     * @param expiryOf The time at which fname expires.
+     */
+    struct RegistrationMetadata {
+        address recoveryOf;
```

Comment:
the attributes can just be called recovery, expiryTs, recoveryDestination, recoveryTs since they are no longer map names

Reply:
so I hesitated to refactor variable names because I would need to update the tests, but I went ahead and implemented the variable refactoring as you suggested (updated the tests).

Honestly the view functions I added were also to make it easier to avoid changing the tests too much. I can remove those view functions and refactor the tests further. The view functions are redundant, information about the metadata of fnames can be obtained directly from the public mapping without helper view functions. The contract size is <23 kb, so the size is OK, but removing the view functions can save some contract size if other functionality is desired (24.576 kb [limit](https://ethereum.org/en/developers/tutorials/downsizing-contracts-to-fight-the-contract-size-limit/))

I also needed to change some function param names from `recovery` to `_recovery` to avoid a collision with the metadata name eg:

`registrationMetadataOf[tokenId].recovery = _recovery;`

This isn't an error, but solhint doesn't like it if it were written as,

`registrationMetadataOf[tokenId].recovery = recovery;`

Reply:
1. removing the helpers will make it easier to ship #109, so i agree about doing it 
2. thanks for updating the names! i know its a pain but it improves readability

Reply:
i removed the vie funcitons and refactored the tests, saves like 300 bytes in deployed contract size